src/include/function/hash/hash_functions.h

```diff
@@ -133,31 +134,52 @@ inline void Hash::operation(
 template<>
 inline void Hash::operation(
     const double& key, common::hash_t& result, common::ValueVector* /*keyVector*/) {
-    result = std::hash<double>()(key);
+    // 0 and -0 are not byte-equivalent, but should have the same hash
+    if (key == 0) {
```

Comment:
I wonder if adding 0.0 can also solve the "0 and -0 are not byte-equivalent" problem without an `if/else` branch.
```
auto newKey = key + 0.0;
result = murmurhash64(*reinterpret_cast<const uint64_t*>(&newKey));
```

Reply:
That does appear to also work.